tests/common.py

```diff
@@ -443,9 +443,9 @@ def compare_forces(
             if compute_du_dx:
                 self.assert_equal_vectors(np.array(ref_du_dx), np.array(test_du_dx), rtol)
             if compute_du_dl:
-                np.testing.assert_almost_equal(ref_du_dl, test_du_dl, rtol)
+                np.testing.assert_allclose(ref_du_dl, test_du_dl, rtol=rtol)
             if compute_du_dp:
-                np.testing.assert_almost_equal(ref_du_dp, test_du_dp, rtol)
+                np.testing.assert_allclose(ref_du_dp, test_du_dp, rtol=rtol, atol=atol)
```

Comment:
nit: `isclose(a, b, rtol=rtol) != isclose(b, a, rtol=rtol)` in general -- the order of `ref_` and `test_` [should probably](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html) be flipped in a few places

Reply:
Since this occurs in other files, and to not block the bug fix, I will address these all together in an upcoming PR